version/version.go

```diff
@@ -87,3 +94,67 @@ func Handler() http.Handler {
 		enc.Encode(v)
 	})
 }
+
+// versionMultiplier consts are used to create a comparable and reversible integer version from a semver string
+// by using 1 million, 1 thousand, and 1 for each part we avoid collisions as long as all parts are less than 1 thousand
+const (
+	majorVersionMultiplier int = 1000000
+	minorVersionMultiplier int = 1000
+	patchVersionMultiplier int = 1
+)
+
+// VersionNum parses the semver version string to look for only the major.minor.patch portion,
+// splits that into 3 parts (disregarding any extra portions), and applies a multiplier to each
+// part to generate a total int value representing the semver.
+// note that this will generate a sortable, and reversible integer as long as all parts remain less than 1000
+// This is currently intended for use in generating comparable versions to set within windows registry entries,
+// allowing for an easy "upgrade-only" detection configuration within intune.
+// Zero is returned for any case where the version cannot be reliably translated
+func VersionNum() int {
+	semverMatch := semverRegexp.FindStringSubmatch(version)
```

Comment:
This will make it impossible to differentiate between different builds (shas) of a given patch. Maybe that's okay

Reply:
yeah I couldn't think of a clean way to handle that, and given the intention felt like this would be alright

Reply:
going to merge this for now but happy to iterate if you have any ideas here!

Reply:
unix timestamp  :)

I forget... Did the windows functionality include lexigraphical string sorts? I assume not